specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,190 @@
+# Localization for Labels in Nimble Components
+
+## Problem Statement
+
+We want to have consistent guidance for when & how to use labels for accessible text inside Nimble components. We also need a plan for how to support localization to non-English languages for those labels.
+
+## Links To Relevant Work Items and Reference Material
+
+[#1090: Unified strategy for providing localized title and aria label for icons within components](https://github.com/ni/nimble/issues/1090)
+
+[PR 1257: Runtime Configuration HLD](https://github.com/ni/nimble/pull/1257)
+
+## Accessible Labels
+For many Nimble components, we already have a sufficient strategy for accessible labels. Guidance for specific components is documented in [nimble-components/docs/accessibility.md](../../packages/nimble-components/docs/accessibility.md).
+
+Issue #1090 primarily covers the various icons used in Nimble components. A summary of our plan for accessible labels for icons currently used in Nimble components is:
+- Increment/decrement buttons in `nimble-number-field` ([#617](https://github.com/ni/nimble/issues/617)): These are `nimble-button` with `content-hidden`, and "Increment" and "Decrement" text. That approach is sufficient, but the text needs localization.
+- `nimble-banner` and `nimble-tooltip` severity icons: No accessible label specifically for the severity icons is currently planned. The banner/tooltip text should usually be sufficient to indicate if the message is an error or informational, so additional "Warning" or "Error" labels would usually be redundant. For the banner, the accessible text comes from the `title`. If we need to specifically support high-severity errors that we want to call the user's attention to, we could consider using the ARIA `role=alert` in that case.
+- `nimble-banner` dismiss button icon: `nimble-button` with `content-hidden`, text comes from `dismissButtonLabel`, with a fallback as "Close". That approach is sufficient. `dismissButtonLabel` should be localized by the clients, the "Close" fallback text is Nimble-provided and needs localization.
```

Comment:
Correct. The "Plan for Client-Provided Labels" section later in the HLD says that we could probably remove the attribute (since generally this one should be shared), I can add a note here too though.